servers/quarkus-server/src/main/java/org/projectnessie/server/authz/CelAccessChecker.java

```diff
@@ -0,0 +1,186 @@
+      throws ForbiddenException {
+    canPerformOpOnReference(context, ref, AuthorizationRuleType.COMMIT_CHANGE_AGAINST_REFERENCE);
+  }
+
+  @Override
+  public void canReadEntityValue(AccessContext context, NamedRef ref, ContentsKey key)
+      throws ForbiddenException {
+    canPerformOpOnPath(context, key, AuthorizationRuleType.READ_ENTITY_VALUE);
+  }
+
+  @Override
+  public void canUpdateEntity(AccessContext context, NamedRef ref, ContentsKey key)
+      throws ForbiddenException {
+    canPerformOpOnPath(context, key, AuthorizationRuleType.UPDATE_ENTITY);
+  }
+
+  @Override
+  public void canDeleteEntity(AccessContext context, NamedRef ref, ContentsKey key)
+      throws ForbiddenException {
+    canPerformOpOnPath(context, key, AuthorizationRuleType.DELETE_ENTITY);
+  }
+
+  private void canPerformOpOnReference(
+      AccessContext context, NamedRef ref, AuthorizationRuleType type) {
+    if (!config.enabled()) {
+      return;
+    }
+    boolean allowed =
+        config.rules().entrySet().stream()
+            .anyMatch(
+                entry -> {
+                  try {
+                    return CELUtil.SCRIPT_HOST
+                        .buildScript(entry.getValue().replace("\"", ""))
+                        .withContainer(CELUtil.CONTAINER)
+                        .withDeclarations(CELUtil.AUTHORIZATION_RULE_DECLARATIONS)
+                        .build()
+                        .execute(
+                            Boolean.class,
+                            ImmutableMap.of(
+                                "ref",
+                                ref.getName(),
+                                "role",
+                                getRoleName(context),
+                                "op",
+                                type.name()));
+                  } catch (ScriptException e) {
+                    throw new ForbiddenException(
+                        String.format(
+                            "Failed to compile query expression with id '%s' and expression '%s' due to: %s",
+                            entry.getKey(), entry.getValue(), e.getMessage()));
+                  }
+                });
+    if (!allowed) {
+      throw new ForbiddenException(
+          String.format(
+              "'%s' is not allowed for Role '%s' on Reference '%s'",
+              type, getRoleName(context), ref.getName()));
+    }
+  }
+
+  private String getRoleName(AccessContext context) {
+    return null != context.user() ? context.user().getName() : "";
+  }
+
+  private void canPerformOpOnPath(
+      AccessContext context, ContentsKey contentsKey, AuthorizationRuleType type) {
+    if (!config.enabled()) {
+      return;
+    }
+    boolean allowed =
+        config.rules().entrySet().stream()
+            .anyMatch(
+                entry -> {
+                  try {
+                    return CELUtil.SCRIPT_HOST
+                        .buildScript(entry.getValue().replace("\"", ""))
```

Comment:
Why does it need to replace here at all? The map should contain the just script IMO.

Reply:
the rule string is wrapped in `"` when we define it in the config `nessie.server.authorization.rules.allow_branch_creation="<rule>"`. Technically I think it would work without wrapping it in `"` but defining`nessie.server.authorization.rules.allow_branch_creation=<rule>` looks a bit weird